node/src/components/linear_chain/state.rs

```diff
@@ -58,17 +69,28 @@ impl LinearChain {
         protocol_version: ProtocolVersion,
         auction_delay: u64,
         unbonding_delay: u64,
+        finality_threshold_fraction: Ratio<u64>,
+        next_upgrade_activation_point: Option<ActivationPoint>,
     ) -> Self {
         LinearChain {
             latest_block: None,
+            key_block_info: Default::default(),
             pending_finality_signatures: PendingSignatures::new(),
             signature_cache: SignatureCache::new(),
             protocol_version,
             auction_delay,
             unbonding_delay,
+            next_upgrade_activation_point,
+            finality_threshold_fraction,
         }
     }
 
+    /// Handles registering an upgrade activation point.
+    pub(super) fn got_upgrade_activation_point(&mut self, activation_point: ActivationPoint) {
+        debug!("got {}", activation_point);
```

Comment:
Wouldn't something along these lines be better?
```suggestion
        debug!(?activation_point, "got an activation point");
```